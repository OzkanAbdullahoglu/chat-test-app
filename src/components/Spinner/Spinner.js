/* eslint-disable linebreak-style */
import styled, { keyframes } from 'styled-components';

const spin = keyframes`
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
`;

export const Spinner = styled.div`
  border: 0.2em solid rgba(0, 0, 0, 0.1);
  border-top: 0.2em solid #00bcd4;
  border-radius: 50%;
  width: 2.2rem;
  height: 2.2rem;
  animation: ${spin} 0.6s linear infinite;
  position: fixed;
  top: 50%;
  bottom: 50%;
  left: 50%;
`;

export default Spinner;
