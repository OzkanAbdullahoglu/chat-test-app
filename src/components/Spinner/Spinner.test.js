/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import Spinner from './Spinner';

describe('Spinner', () => {
  it('renders without crashing', () => {
    const spinner = shallow(<Spinner />);
    expect(toJson(spinner)).toMatchSnapshot();
  });
});
