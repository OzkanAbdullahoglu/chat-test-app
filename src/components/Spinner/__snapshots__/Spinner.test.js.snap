// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders without crashing 1`] = `
.c0 {
  border: 0.2em solid rgba(0,0,0,0.1);
  border-top: 0.2em solid #00bcd4;
  border-radius: 50%;
  width: 2.2rem;
  height: 2.2rem;
  -webkit-animation: Hqxfx 0.6s linear infinite;
  animation: Hqxfx 0.6s linear infinite;
  position: fixed;
  top: 50%;
  bottom: 50%;
  left: 50%;
}

<div
  className="c0"
/>
`;
